File: php/1_Two_Sum.php
<?php

class Solution
{

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target)
    {
        $arNumsSwap = [];
        $arResult = [];
        foreach ($nums as $key => $num) {
            if (isset($arNumsSwap[$target - $num])) {
                $arResult = [$arNumsSwap[$target - $num], $key];
            } else {
                $arNumsSwap[$num] = $key;
            }
        }

        return $arResult;
    }

}

File: python/1_Two_Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        num_map = {}
        for i, num in enumerate(nums):
            if target - num in num_map:
                return [num_map[target - num], i]
            else:
                num_map[nums[i]] = i
        return [-1, -1]
